inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -17,55 +17,241 @@
 import de.tudarmstadt.ukp.inception.recommendation.api.RecommendationService;
 import de.tudarmstadt.ukp.inception.recommendation.api.model.Preferences;
+import de.tudarmstadt.ukp.inception.recommendation.log.RecommenderEvaluationResultEventAdapter.Details;
+
+public class RecommendationSidebar extends AnnotationSidebar_ImplBase {
+	private static final long serialVersionUID = 4306746527837380863L;
+
+	private Logger log = LoggerFactory.getLogger(getClass());
+	private WebComponent chartContainer;
+
+	private @SpringBean RecommendationService recommendationService;
+	private @SpringBean EventRepository eventRepo;
+
+	IModel<AnnotatorState> aModel;
+
+	int maxPointsToPlot = 50;
+
+	public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel, AnnotationActionHandler aActionHandler,
+			JCasProvider aJCasProvider, AnnotationPage aAnnotationPage) {
+		super(aId, aModel, aActionHandler, aJCasProvider, aAnnotationPage);
+		this.aModel = aModel;
+		IModel<Preferences> model = LambdaModelAdapter.of(
+				() -> recommendationService.getPreferences(aModel.getObject().getUser(),
+						aModel.getObject().getProject()),
+				(v) -> recommendationService.setPreferences(aModel.getObject().getUser(),
+						aModel.getObject().getProject(), v));
+
+		Form<Preferences> form = new Form<>("form", CompoundPropertyModel.of(model));
+
+		form.add(new NumberTextField<Integer>("maxPredictions", Integer.class).setMinimum(1).setMaximum(10).setStep(1));
+
+		form.add(new CheckBox("showAllPredictions"));
+
+		form.add(new LambdaAjaxButton<>("save", (_target, _form) -> aAnnotationPage.actionRefreshDocument(_target)));
+
+		add(form);
+
+		chartContainer = new Label("chart-container");
+		chartContainer.setOutputMarkupId(true);
+		add(chartContainer);
+	}
+
+	@Override
+	public void renderHead(IHeaderResponse aResponse) {
+		super.renderHead(aResponse);
+
+		// import Js
+		aResponse.render(JavaScriptHeaderItem.forReference(new WebjarsJavaScriptResourceReference("c3/current/c3.js")));
+		aResponse.render(
+				JavaScriptHeaderItem.forReference(new WebjarsJavaScriptResourceReference("d3js/current/d3.js")));
+
+		// import Css
+		aResponse.render(CssHeaderItem.forReference(new WebjarsCssResourceReference("c3/current/c3.css")));
+	}
+
+	@OnEvent
+	public void onRenderAnnotations(RenderAnnotationsEvent aEvent) {
+		try {
+			log.info("rendered annotation event");
+
+			StringBuilder chartType = new StringBuilder();
+			
+			HashMap<String, List<Double>> recommenderScoreMap = getLatestScores(aEvent);
+
+			// iterate over the recommender score map to create data arrays to feed to the
+			// c3 graph
+			StringBuilder dataColumns = new StringBuilder();
+			Iterator<Entry<String, List<Double>>> it = recommenderScoreMap.entrySet().iterator();
+			while (it.hasNext()) {
+				Entry<String, List<Double>> pair = it.next();
+
+				String jsonString = toJsonString(pair);
+				jsonString = jsonString.replaceAll("\".*?\"", "");
```

Comment:
It would be nice if you could comment in the code why you need to replace stuff here.